spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/functions/BucketFunction.java

```diff
@@ -128,6 +133,23 @@ public String name() {
     public DataType resultType() {
       return DataTypes.IntegerType;
     }
+
+    public int gcd(int num1, int num2) {
```

Comment:
nit: a, b for brevity?  Not sure if num1, num2 carries any more extra information